include/onnxruntime/core/session/onnxruntime_cxx_api.h

```diff
@@ -348,25 +400,50 @@ struct ModelMetadata : Base<OrtModelMetadata> {
   int64_t GetVersion() const;
 };
 
+/** \brief Wrapper around ::OrtSession
+*
+*/
 struct Session : Base<OrtSession> {
-  explicit Session(std::nullptr_t) {}
+  explicit Session(std::nullptr_t) {} ///< \brief Create an empty Session object, must be assigned a valid one to be used 
   Session(Env& env, const ORTCHAR_T* model_path, const SessionOptions& options);
   Session(Env& env, const ORTCHAR_T* model_path, const SessionOptions& options, OrtPrepackedWeightsContainer* prepacked_weights_container);
   Session(Env& env, const void* model_data, size_t model_data_length, const SessionOptions& options);
 
-  // Run that will allocate the output values
+  /** \brief Run the model returning results in an Ort allocated vector.
+  * 
+  * The caller provides a list of inputs and a list of the desired outputs to return.
```

Comment:
I've had this question and haven't found the answer yet, perhaps it would be good to document:
how are multiple, identical input/output names handled (e.g., input names = ["a", "b", "a"])? what if such inputs have different values provided?

Reply:
I don't know.. I'd expect it to be an error or undefined behavior. We'd have to try it and see, then comment.